docs/types/reference-types.rst

```diff
@@ -336,6 +336,74 @@ Array Members
         }
     }
 
+.. index:: ! array;slice
+
+.. _array-slices:
+
+Array Slices
+------------
+
+
+Array slices are a view on a contiguous portion of an array.
+They are written as ``x[start:end]``, where ``start`` and
+``end`` are expressions resulting in a uint256 type (or
+implicitly convertible to it). The first element of the
+slice is ``x[start]`` and the last element is ``x[end - 1]``.
+
+If ``start`` is greater than ``end`` or if ``end`` is greater
+than the length of the array, an exception is thrown.
+
+Both ``start`` and ``end`` are optional: ``start`` defaults
+ to ``0`` and ``end`` defaults to the length of the array.
+
+Array slices do not have any members. They are implicitly
+convertible to arrays of their underlying type
+and support index access. Index access is not absolute
+in the underlying array, but relative to the start of
+the slice.
+
+Array slices do not have a type name which means
+no variable can have an array slices as type,
+they only exist in intermediate expressions.
+
+.. note::
+    As of now, array slices are only implemented for calldata arrays.
+
+Array slices are useful to ABI-decode secondary data passed in function parameters:
+
+::
+
+    pragma solidity >=0.4.99 <0.7.0;
+
+    contract C {
+        address owner;
+
+        modifier atleastPayloadSize(uint size) {
+            require(msg.data.length >= size + 4);
+            _;
+        }
+
+        modifier onlySelf() {
+            require(msg.sender == address(this));
+            _;
+        }
+
+        function setOwner(address _owner) onlySelf public {
+            owner = _owner;
+        }
+
+        function forward(bytes calldata _payload) atleastPayloadSize(20) external {
+            bytes4 sig = abi.decode(_payload[:4], (bytes4));
```

Comment:
Is this the only way to convert?

Reply:
If I do

```
bytes4 sig = bytes4(_payload[:4]);
```

I get the following error

```
Error: Explicit type conversion not allowed from "bytes calldata slice" to "bytes4".
...
```